src/Microsoft.AspNetCore.DataProtection/Internal/DockerUtils.cs

```diff
@@ -0,0 +1,99 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Runtime.InteropServices;
+
+namespace Microsoft.AspNetCore.DataProtection.Internal
+{
+    internal static class DockerUtils
+    {
+        private static Lazy<bool> _isDocker = new Lazy<bool>(IsProcessRunningInDocker);
+
+        public static bool IsDocker => _isDocker.Value;
+
+        public static bool IsVolumeMountedFolder(DirectoryInfo directory)
```

Comment:
We are doing tons of file operations and parsing, I would prefer peoples apps not crashing if they have slightly invalid fstab. It's better to wrap this thing in try catch.